Library/Homebrew/test/livecheck/livecheck_spec.rb

```diff
@@ -0,0 +1,160 @@
+      url "https://brew.sh/test-0.0.1.tgz"
+    end
+  end
+
+  let(:args) { double("livecheck_args", full_name?: false, json?: false, quiet?: false, verbose?: true) }
+
+  describe "::formula_name" do
+    it "returns the name of the formula" do
+      expect(livecheck.formula_name(f, args: args)).to eq("test")
+    end
+
+    it "returns the full name" do
+      allow(args).to receive(:full_name?).and_return(true)
+
+      expect(livecheck.formula_name(f, args: args)).to eq("test")
+    end
+  end
+
+  describe "::status_hash" do
+    it "returns a hash containing the livecheck status" do
+      expect(livecheck.status_hash(f, "error", ["Unable to get versions"], args: args))
+        .to eq({
+                 formula:  "test",
+                 status:   "error",
+                 messages: ["Unable to get versions"],
+                 meta:     {
+                   livecheckable: true,
+                 },
+               })
+    end
+  end
+
+  describe "::skip_conditions" do
+    it "skips a deprecated formula without a livecheckable" do
+      expect { livecheck.skip_conditions(f_deprecated, args: args) }
+        .to output("test_deprecated : deprecated\n").to_stdout
+        .and not_to_output.to_stderr
+    end
+
+    it "skips a versioned formula without a livecheckable" do
+      expect { livecheck.skip_conditions(f_versioned, args: args) }
+        .to output("test@0.0.1 : versioned\n").to_stdout
+        .and not_to_output.to_stderr
+    end
+
+    it "skips a HEAD-only formula if not installed" do
+      expect { livecheck.skip_conditions(f_head_only, args: args) }
+        .to output("test_head_only : HEAD only formula must be installed to be livecheckable\n").to_stdout
+        .and not_to_output.to_stderr
+    end
+
+    it "skips a formula with a GitHub Gist stable URL" do
+      expect { livecheck.skip_conditions(f_gist, args: args) }
+        .to output("test_gist : skipped - Stable URL is a GitHub Gist\n").to_stdout
+        .and not_to_output.to_stderr
+    end
+
+    it "skips a formula with a skip livecheckable" do
+      expect { livecheck.skip_conditions(f_skip, args: args) }
+        .to output("test_skip : skipped - Not maintained\n").to_stdout
+        .and not_to_output.to_stderr
+    end
+
+    it "returns false for a non-skippable formula" do
+      expect(livecheck.skip_conditions(f, args: args)).to eq(false)
+    end
+  end
+
+  describe "::checkable_urls" do
+    it "returns the list of URLs to check" do
+      expect(livecheck.checkable_urls(f))
+        .to eq(
+          ["https://github.com/Homebrew/brew.git", "https://brew.sh/test-0.0.1.tgz", "https://brew.sh"],
+        )
+    end
+  end
+
+  describe "::preprocess_url" do
+    let(:url) { "https://github.s3.amazonaws.com/Homebrew/brew/releases/latest" }
+
+    it "returns the preprocessed URL for livecheck to use" do
+      expect(livecheck.preprocess_url(url))
+        .to eq("https://github.com/Homebrew/brew/releases/latest")
```

Comment:
```suggestion
    let(:url) { "https://github.s3.amazonaws.com/downloads/Homebrew/brew/1.0.0.tar.gz" }

    it "returns the preprocessed URL for livecheck to use" do
      expect(livecheck.preprocess_url(url))
        .to eq("https://github.com/Homebrew/brew.git")
```

The `github.s3.amazonaws` URLs (found in the `syck` and `wrk-trello` formulae) follow a format like `https://github.s3.amazonaws.com/downloads/Homebrew/brew/1.0.0.tar.gz`. It may be more representative to use this URL, as we'll probably never see one ending in `.../releases/latest`.

This also exercises a little more of the code, since `/releases/latest` URLs are a special case and don't get converted into the `.git` URL. I'll probably expand this later but this may be better for right now.